Unable To Reference Custom Parameter Definitions

I have a base "swagger.yml" defined as:

```yaml
swagger: 2.0
info:
  title: V API
  description: MyServer
  version: 0.7.5
host: "XXX"
schemes:
  - "http"
  - "https"
consumes:
  - "application/json"
produces:
  - "application/json"
parameters:
  domain:
    name: "X-VirtualDomain"
    in: header
    description: The domain (account) you're attempting to interact with.
    required: true
    type: string
```

When I try to reference the parameter, I get JSON errors.

```raw
###
#   summary: does initial signup for credentials
#   produces:
#     - application/json
#   parameters:
#     - $ref: "#/parameters/domain"
#     - name: email
#       in: body
#       description: The email address to associate with the new account.
#       required: true
#       type: string
#       format: email
#     - name: password
#       in: body
#       description: The password for the new account.
#       required: true
#       type: password
#   responses:
#     200:
#       description: User Created
#       schema:
#         headers:
#           X-Auth-Token:
#             description: An auth token issued to this user
#             type: string
#
#
###
POST    /signup                                    controllers.rest.CredentialsController.signUp


play.api.http.HttpErrorHandlerExceptions$$anon$1: Execution exception[[JsResultException: JsResultException(errors:List((,List(ValidationError(List('name' is undefined on object: {"$ref":"#/parameters/domain"}),WrappedArray())))))]]
        at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:280)
        at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:206)
        at play.api.GlobalSettings$class.onError(GlobalSettings.scala:160)
        at play.api.DefaultGlobal$.onError(GlobalSettings.scala:188)
        at play.api.http.GlobalSettingsHttpErrorHandler.onServerError(HttpErrorHandler.scala:98)
        at play.core.server.netty.PlayRequestHandler$$anonfun$2$$anonfun$apply$1.applyOrElse(PlayRequestHandler.scala:100)
        at play.core.server.netty.PlayRequestHandler$$anonfun$2$$anonfun$apply$1.applyOrElse(PlayRequestHandler.scala:99)
        at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:344)
        at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:343)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
Caused by: play.api.libs.json.JsResultException: JsResultException(errors:List((,List(ValidationError(List('name' is undefined on object: {"$ref":"#/parameters/domain"}),WrappedArray())))))
        at play.api.libs.json.JsReadable$$anonfun$2.apply(JsReadable.scala:23)
        at play.api.libs.json.JsReadable$$anonfun$2.apply(JsReadable.scala:23)
        at play.api.libs.json.JsResult$class.fold(JsResult.scala:73)
        at play.api.libs.json.JsError.fold(JsResult.scala:13)
        at play.api.libs.json.JsReadable$class.as(JsReadable.scala:21)
        at play.api.libs.json.JsUndefined.as(JsLookup.scala:137)
        at com.iheart.playSwagger.SwaggerSpecGenerator$$anonfun$mergeByName$2.apply(SwaggerSpecGenerator.scala:203)
        at com.iheart.playSwagger.SwaggerSpecGenerator$$anonfun$mergeByName$2.apply(SwaggerSpecGenerator.scala:202)
        at scala.collection.TraversableLike$$anonfun$filterImpl$1.apply(TraversableLike.scala:259)
        at scala.collection.immutable.List.foreach(List.scala:381)
```

--- a/src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala
+++ b/src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala
@@ -200,8 +200,9 @@ final case class SwaggerSpecGenerator(
       val name = (bs \ "name").as[String]
       findByName(toMerge, name).fold(bs) { f ⇒ bs.as[JsObject] deepMerge f }
     } ++ toMerge.value.filter { tm ⇒
-      val name = (tm \ "name").as[String]
-      findByName(base, name).isEmpty
+      (tm \ "name").validate[String].fold({ errors ⇒ true }, { name ⇒
+        findByName(base, name).isEmpty
+      })
     })
   }
 

--- a/src/test/resources/referencing.routes
+++ b/src/test/resources/referencing.routes
@@ -1,0 +1,11 @@
+
+###
+#  summary: Echos the magic reference parameter
+#  parameters:
+#    - $ref: "#/parameters/magic"
+#    - name: notMagic
+#      description: This is not an important parameter
+###
+POST  /magic/echoMagic/:type        com.iheart.playSwagger.test.echoMagic(type, notMagic)
+
+

--- a/src/test/resources/test.routes
+++ b/src/test/resources/test.routes
@@ -11,6 +11,8 @@
 
 ->   /aCustomerRouter      com.myCompany.Router
 
+->   /references           referencing.Routes
+
 GET  /abc                  com.iheart.Application.test()
 
 ###

--- a/src/test/scala/com/iheart/playSwagger/DefinitionGeneratorSpec.scala
+++ b/src/test/scala/com/iheart/playSwagger/DefinitionGeneratorSpec.scala
@@ -86,7 +86,7 @@ class DefinitionGeneratorSpec extends Specification {
       result.properties.last.referenceType === None
     }
 
-    "read seqence items" >> {
+    "read sequence items" >> {
       val result = DefinitionGenerator("com.iheart").definition("com.iheart.playSwagger.FooWithSeq")
       result.properties.head.items.get.referenceType === Some("com.iheart.playSwagger.SeqItem")
     }

--- a/src/test/scala/com/iheart/playSwagger/SwaggerSpecGeneratorSpec.scala
+++ b/src/test/scala/com/iheart/playSwagger/SwaggerSpecGeneratorSpec.scala
@@ -1,7 +1,7 @@
 package com.iheart.playSwagger
 
 import org.specs2.mutable.Specification
-import play.api.libs.json.{JsValue, JsArray, JsObject}
+import play.api.libs.json.{JsArray, JsObject, JsString, JsValue}
 
 case class Track(name: String, genre: Option[String], artist: Artist, related: Seq[Artist], numbers: Seq[Int])
 case class Artist(name: String, age: Int)
@@ -36,7 +36,7 @@ class SwaggerSpecGeneratorSpec extends Specification {
     "avoid double slash at the middle" >> {
       gen.fullPath("p/", "/d") === "/p/d"
     }
-    
+
     "allow internal double slash" >> {
       gen.fullPath("p/", "/d//c") === "/p/d//c"
     }
@@ -287,6 +287,18 @@ class SwaggerSpecGeneratorIntegrationSpec extends Specification {
       val properties = (definitionsJson \ "com.iheart.playSwagger.FooWithSeq2" \ "properties").as[JsObject]
       (properties \ "abc1" \ "items" \ "type").as[String] === "string"
       (properties \ "abc2" \ "items" \ "items" \ "type").as[String] === "integer"
+    }
+
+    "accept parameter references" >> {
+      val parameters = (pathJson \ "/references/magic/echoMagic/{type}" \ "post" \ "parameters").as[Seq[JsObject]]
+
+      parameters must contain((entry: JsObject) ⇒
+        entry.value.get("$ref").contains(JsString("#/parameters/magic")))
+        .exactly(1.times)
+
+      parameters must contain((entry: JsObject) ⇒
+        entry.value.get("name").contains(JsString("notMagic")))
+        .exactly(1.times)
     }
 
     // TODO: routes order

